man/systemd.exec.xml

```diff
@@ -1242,28 +1242,63 @@ RestrictNamespaces=~cgroup net</programlisting>
         stopped. This setting is implied if <varname>DynamicUser=</varname> is set.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>PrivateMounts=</varname></term>
+
+        <listitem><para>Takes a boolean parameter. If set, the processes of this unit will be run in their own private
+        file system (mount) namespace with all mount propagation from the processes towards the host's main file system
+        namespace turned off. This means any file system mount points established or removed by the unit's processes
+        will be private to them and not be visible to the host. However, file system mount points established or
+        removed on the host will be propagated to the unit's processes. See <citerefentry
+        project='man-pages'><refentrytitle>mount_namespaces</refentrytitle><manvolnum>7</manvolnum></citerefentry> for
+        details on file system namespaces. When turned on, this executes three operations for each invoked process: a
+        new <constant>CLONE_NEWNS</constant> namespace is created, after which all existing mounts are remounted to
+        <constant>MS_SLAVE</constant> to disable propagation from the host to the unit's processes (but leaving
+        propagation in the opposite direction in effect). Finally, the mounts are remounted again to the propagation
+        mode configured with <varname>MountFlags=</varname>, see below. Note that the other file system namespace unit
+        settings (such as <varname>PrivateMounts=</varname>, <varname>PrivateTmp=</varname>,
+        <varname>PrivateDevices=</varname>, <varname>ProtectSystem=</varname>, <varname>ProtectHome=</varname>,
+        <varname>ReadOnlyPaths=</varname>, <varname>InaccessiblePaths=</varname>, <varname>ReadWritePaths=</varname>,
+        …) have the effect of implicitly turning on file system mount namespacing in an equivalent fashion: this option
+        is hence primarily useful to explicitly request this behaviour if no other file system namespace settings are
+        used that request it implicitly. Note that file system namespaces are set up individually for each process
+        forked off by the service manager. Mounts established in the namespace of the process created by
+        <varname>ExecStartPre=</varname> will hence be cleaned up automatically as soon as that process exits and will
+        not be available to subsequent processes forked off for <varname>ExecStart=</varname> (and similar applies to
+        the various other commands configured for units). Defaults to off.</para></listitem>
+      </varlistentry>
```

Comment:
it might be nice to mention JoinNamespaceOf somewhere... maybe ?

Reply:
Also, that last behaviour seems strange. wouldn't it be more logical to have ExecStartPre share the namespace unless explicitely told not to (+ prefix in ExecStartPre or PermissionStartOnly)

that might be for a different PR, though

Reply:
will add a comment about JoinsNamespaceOf=